run_pacta_data_preparation.R

```diff
@@ -504,20 +504,38 @@ iss_company_emissions <-
 
 logger::log_info("Formatting and saving file: \"iss_entity_emission_intensities.rds\".")
 
+factset_financial_data <-
+  readRDS(factset_financial_data_path) %>%
+  select(factset_entity_id,issue_type, adj_price, adj_shares_outstanding) %>%
+  filter(!is.na(factset_entity_id) & !is.na(adj_shares_outstanding)) %>%
+  filter(issue_type %in% c("EQ", "PF", "CP")) %>%
+  summarize(
+    mkt_val = sum(adj_price * adj_shares_outstanding, na.rm = TRUE),
+    .by = "factset_entity_id"
+  )
+
+factset_entity_info <-
+  readRDS(factset_entity_info_path) %>%
+  select(factset_entity_id, iso_country, sector_code, factset_sector_desc) %>%
+  left_join(factset_financial_data, by = "factset_entity_id")
+
+rm(factset_financial_data)
+invisible(gc())
+
 iss_entity_emission_intensities <-
   readRDS(factset_entity_financing_data_path) %>%
+  left_join(factset_entity_info, by = "factset_entity_id") %>%
+  filter(countrycode::countrycode(iso_country, "iso2c", "iso4217c") == currency) %>%
   left_join(currencies, by = "currency") %>%
   mutate(
-    ff_mkt_val = ff_mkt_val * exchange_rate,
-    ff_debt = ff_debt * exchange_rate,
+    ff_mkt_val = if_else(!is.na(mkt_val), mkt_val, NA_real_),
+    ff_debt = if_else(!is.na(ff_debt), ff_debt * exchange_rate, NA_real_),
     currency = "USD"
   ) %>%
-  select(-exchange_rate) %>%
-  group_by(factset_entity_id, currency) %>%
   summarise(
-    ff_mkt_val = sum(ff_mkt_val, na.rm = TRUE),
-    ff_debt = sum(ff_debt, na.rm = TRUE),
-    .groups = "drop"
+    ff_mkt_val = mean(ff_mkt_val, na.rm = TRUE),
+    ff_debt = mean(ff_debt, na.rm = TRUE),
```

Comment:
Why are we talking the `mean` here?

Reply:
better answered by @Antoine-Lalechere, but I think because there are slight variations between the same entity, so this is meant to average them out

Reply:
exactly - there are around 150 securities for which the entity's debt may differ by a maximum of 0.1% between two securities for 2023Q4. Those data are often selected from the same timestamps it's hard to figure out where the rounding issue come from, and. that's why given the size of the issue, it has been decided to take mean here